### DISCUSSION TITLE
Dynamic OPML

### DISCUSSION THREAD

**Discussion by stanthewizzard** [FreshRSS/FreshRSS#4769] [Q&A] [answered]
Hello

I'm a bit lost with dynamic OPML.
Could I use DOPML with feedly (for ex) and have FreshRSS sync with it ?

Thanks for help

**Comment by Alkarex** [comment, marked as answer]
FreshRSS supports continuously exporting and importing OPML (making it dynamic), in a similar way than an RSS/ATOM feed, meaning that the side exporting it must make it available as an HTTP resource with an URL.

For your scenario to work, the other side needs also to support dynamic OPML. Have you seen anything of this type in Feedly?

Note that FreshRSS also supports exporting and importing RSS/ATOM, for instance to re-export a feed, category, user label...

See our documentation about user queries and how to share them, in particular the OPML section:
https://freshrss.github.io/FreshRSS/en/users/user_queries.html#share-your-user-queries

**Reply by tobiasholle** [reply]
Hi! I installed FreshRSS and it's great! I want to share my Feeds which are in various categories with my colleagues who can then choose beetween different categories to subscribe to them to also receive the Feeds. So basically I just want to provide different categories that they can subscribe to (by adding the link ?) to their own RSS Reader and this should be dynamic - if I change Feeds in the category, it should be changed in the category as well. So the Feeds they subscribe to should not go through my FreshRSS, only the information which Feeds are in the category.
Is this possible with FreshRSS and if yes how? I'm not sure if Dynamic OPML is the right approach for this... would be great if anyone could help!

**Reply by Alkarex** [reply]
@tobiasholle See our documentation about user queries and how to share them, in particular the OPML section:
https://freshrss.github.io/FreshRSS/en/users/user_queries.html#share-your-user-queries

For instance, if you share one of your categories this way, your colleagues can either subscribe to the resulting RSS feed of your instance, or subscribe to the Dynamic OPML to reproduce the different sub-feeds.

**Reply by tobiasholle** [reply]
@Alkarex thank you so much for the quick response. I tried it this way but none of the apps, theoretically able to work with dynamic OPML, worked (I.e. Inoreader or Nextcloud News).
For the RSS Feed, it worked but it only refreshed when I refreshed it in FreshRSS before. So no independency from the server.

**Reply by Alkarex** [reply]
On your side, you should make sure to have a working cron job to get updates automatically (built-in in your use our official Docker image).
On your colleagues side, make sure to have shared the correct link, which should be of the following format `https://freshrss.example.net/api/query.php?user=alice&t=aZ09&f=opml` (and make sure that it returns a valid response)
If your observe a problem when sharing your dynamic OPML with an application supposed to support it, please open a new issue with details (e.g. screenshots, logs).

**Reply by tobiasholle** [reply]
Thank you! Works now but there are not enough RSS Readers supporting dynamic OPML as far as I can see. So working but not really usable for my case... maybe in the future. I asked the developers of NetNewsWire to implement the dynamic OPML import.

**Reply by Alkarex** [reply]
Well, tell your colleagues to use FreshRSS then ;-)

**Reply by tobiasholle** [reply]
Yeah, they wouldn't use it. They need really easy solutions to set up. I'm even sceptical if they will update an OPML File from time to time - that's why I wanted it to update automatically with dynamic OPML. :D

**Reply by Alkarex** [reply]
You can host a FreshRSS instance with multiple accounts for them ;-)

**Reply by tobiasholle** [reply]
Haha. They need an easy app, also for their phones. And I will share the feeds with more people than I can create accounts for and want to. Sorry. :'(

**Reply by tobiasholle** [reply]
One step back to the refreshing: "On your side, you should make sure to have a working cron job to get updates automatically (built-in in your use our official Docker image)." 
I find this page: https://freshrss.github.io/FreshRSS/en/users/09_refreshing_feeds.html#automatic-update-with-cron but From there on I'm lost. I'm no programmer and would probably destroy more if I would start with that here. :D
Is there an easy way to say: Please update/refresh every second minute?

**Reply by Alkarex** [reply]
The easiest is to use our official Docker image. Otherwise it depends on your type of installation. Please open a new question if needed, not keep this one focussed on dynamic OPML
